### PR TITLE
19 method spaceflow

### PR DESCRIPTION
**Pull request by zsfrbkv**
Added a script for SpaceFlow, a `yml` file and a config file.

### CODE REVIEW

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

File: method/SpaceFlow/method_spaceflow.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+use_cuda = torch.cuda.is_available()
+device = 1 if use_cuda else 0
+
+# Filter genes expressed in < 3 cells
+sc.pp.filter_genes(adata, min_cells=3)
+
+# Create a SpaceFlow object and preprocess it
+sf = SpaceFlow.SpaceFlow(adata=adata)
+sf.preprocessing_data(n_top_genes=3000)
```
Comment (niklasmueboe) [inline method/SpaceFlow/method_spaceflow.py:141]: This preprocessing includes quite a few steps. Check if you can preovide preprocessed data already? Then we would take the preprocessing out of this script.

Comment (zsfrbkv) [reply method/SpaceFlow/method_spaceflow.py]: I removed the steps up until PCA and implemented the graph construction implicitly. Seems to work ok. Then later processed data should be passed to the method.

**Review by niklasmueboe — APPROVED** [review]

File: method/SpaceFlow/method_spaceflow.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+from SpaceFlow import SpaceFlow
+import pandas as pd
+import warnings
+import torch
+
+use_cuda = torch.cuda.is_available()
+device = 1 if use_cuda else 0
+
+# Filter genes expressed in < 3 cells
+sc.pp.filter_genes(adata, min_cells=3)
```
Comment (niklasmueboe) [inline method/SpaceFlow/method_spaceflow.py:137]: this should be removed too I guess as it is also preprocessing

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

File: method/SpaceFlow/config.json
Comment (niklasmueboe) [inline method/SpaceFlow/config.json:1]: Missed That before can you add this into a subfolder called `config`

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

**Review by niklasmueboe — APPROVED** [review]

**Pull request closed without merging**